### ISSUE TITLE
Duplicate spanID Gateway trace in Jaeger tracing

### ISSUE THREAD

**Issue by jakubdyszkiewicz** [kumahq/kuma#4833]
### What happened?

Do we send duplicated traces by accident?

![image](https://user-images.githubusercontent.com/5459824/184873829-acb920ff-8ce4-4b3e-8133-9e566767cf7f.png)

**Comment by lahabana** [comment]
More strange things with traces on the Gateway:

<img width="1653" alt="image" src="https://user-images.githubusercontent.com/702355/215494172-8a6d485e-0da4-469c-9301-4ed391c96dd7.png">

<img width="1643" alt="image" src="https://user-images.githubusercontent.com/702355/215494316-bf33298a-01b0-4337-a69d-fc5b306da2af.png">


Some missing tags in these screenshots:

First screen:

```
upstream_cluster | "demo-app_kuma-demo_svc_5000-41c96d8c484e3ae3" |  
-- | -- | --
upstream_cluster.name | "demo-app_kuma-demo_svc_5000-41c96d8c484e3ae3" |  
user_agent | "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
```

Second:

```
upstream_cluster | "localhost:5000" |  
-- | -- | --
upstream_cluster.name | "localhost_5000" |  
user_agent | "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
```

Seems like the app trace has the same span as the gateway...

**Comment by lahabana** [comment]
@michaelbeaumont was this done?

**Comment by michaelbeaumont** [comment]
@lahabana yeah, looks like it wasn't closed because the PR was merged to release-2.3
